packages/app/src/components/UserSettings/InfoCard.tsx

```diff
@@ -32,24 +37,47 @@ export const InfoCard = () => {
     }
   };
 
-  let clipboardText = buildMetadata.title;
-  if (buildMetadata.card?.length) {
+  const buildDetailsTitle = () => {
+    if (buildInfo?.title && Object.keys(buildInfo?.card ?? {}).length > 0) {
+      return buildInfo.title;
+    }
+    if (!buildInfo?.title && Object.keys(buildInfo?.card ?? {}).length > 0) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'Set a custom title for your build details to display your preferred title.',
+      );
+      return 'Build Metadata';
+    }
+    return buildMetadata.title;
+  };
+
+  let clipboardText = buildDetailsTitle();
+  const buildDetails = Object.entries(
+    buildInfo?.card && Object.keys(buildInfo.card).length > 0
+      ? buildInfo.card
+      : buildMetadata?.card || {},
+  ).map(([key, value]) => `${key}: ${value}`);
+  if (buildDetails?.length) {
     clipboardText += '\n\n';
-    buildMetadata.card.forEach(text => {
+    buildDetails.forEach(text => {
       clipboardText += `${text}\n`;
     });
   }
 
-  const filteredCards = showBuildInformation
-    ? buildMetadata.card
-    : buildMetadata.card.filter(
-        text =>
-          text.startsWith('RHDH Version') ||
-          text.startsWith('Backstage Version'),
-      );
+  const filteredContent = () => {
+    if (Object.keys(buildInfo?.card ?? {})?.length > 0) {
+      return buildDetails.slice(0, 2);
+    }
```

Comment:
Isn't this to complex code for simple?

```suggestion
    if (buildInfo?.card) {
      return buildDetails.slice(0, 2);
    }
```